client/src/main/com/sinch/sdk/domains/voice/adapters/VoiceService.java

```diff
@@ -0,0 +1,63 @@
+package com.sinch.sdk.domains.voice.adapters;
+
+import com.sinch.sdk.auth.adapters.VoiceApplicationAuthManager;
+import com.sinch.sdk.core.exceptions.ApiAuthException;
+import com.sinch.sdk.core.http.AuthManager;
+import com.sinch.sdk.core.http.HttpClient;
+import com.sinch.sdk.domains.voice.CalloutsService;
+import com.sinch.sdk.models.Configuration;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+public class VoiceService implements com.sinch.sdk.domains.voice.VoiceService {
+
+  private static final String SECURITY_SCHEME_KEYWORD = "Signed";
+
+  private final Configuration configuration;
+  private final HttpClient httpClient;
+  private CalloutsService callouts;
+
+  private Map<String, AuthManager> clientAuthManagers;
+
+  public VoiceService(Configuration configuration, HttpClient httpClient) {
+
+    // Currently, we are not supporting unified credentials: ensure application credentials are
+    // defined
+    Objects.requireNonNull(configuration.getApplicationKey(), "'applicationKey' cannot be null");
```

Comment:
Maybe we can also check the application key and secret are not empty strings

Reply:
Is it really useful ?
Next step could not null and nor empty and nor empty when spaces or dead characters are stripped from the string.
In this case checking against a null value we are assuming developers had set a value.... and we could expect that the contract is satisfied.

Reply:
Indeed, I was wondering if we could get an glimpse at the application key generator to determine the format, as we do for credit card numbers... but you're right, let's assume that when a developer provides a non-null value, he provides the correct one. If not, the server will be pleased to tell him the credentials are not correct ;)